foil.tex

```diff
@@ -50,6 +50,242 @@ \chapter*{Introduction}
+
+%%
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+\begin{flushleft}
+
+The introduction of airfoil:
+\begin{enumerate}
+    \item
+    Chord ($c$)
+    \item
+    Maximum camber ($m$)
+    \item
+    Maximum camber location ($p$)
+    \item
+    Maximum thickness ($h$)
+    \item
+    Camber line (red dashed line)
+    \item
+    Leading edge ($x = 0, y = 0$)
+    \item
+    Trailing edge ($x = c, y = 0$)
+\end{enumerate}
+\begin{figure}[h]
+    \centering
+    \includegraphics[width=0.5\textwidth]{airfoil_shape_image.eps}
+    \caption{The parameters in wing profile.}
+    \label{fig:airfoil_shape_image}
+\end{figure}
+% figure downloaded from: https://www.researchgate.net/figure/AIRFOIL-SHAPE-PARAMETERS_fig4_267490031
+
+\bigbreak
+The wing profile is defined by four-digit ($XYZZ$):
+\begin{enumerate}
+    \item
+    The 1st digit is the maximum camber as percentage of the chord.
+    \begin{equation}
+        m = c \times (X/100)
+    \end{equation}
+    \item
+    The 2nd digit is the maximum camber location in tenths of the chord.
+    \begin{equation}
+        p = c \times (Y/10)
+    \end{equation}
+    \item
+    The last two digits are the maximum thickness as percent of the chord.
+    \begin{equation}
+        h = c \times (ZZ/100)
+    \end{equation}
+\end{enumerate}
+
+\bigbreak
+The thickness equation is
+\begin{equation}
+    y_h = 5h[0.2969\sqrt{x}-0.1260x-0.3516x^2+0.2843x^3-0.1015x^4]
+\end{equation}
+where $x$ is the position along the chord (from $0$ to $1$), $y_h$ is the
+half thickness at a given value of $x$, and $h$ is the maximum thickness.
+
+\bigbreak
+The camber line equation is
+\begin{equation}
+    y_c =
+    \begin{cases}
+    \frac{m}{p^2} (2px - x^2); & 0 \leq x \leq p \\
+    \frac{m}{(1-p)^2} (1 - 2p + 2px - x^2);  &  p \leq x \leq 1
+    \end{cases}
+\end{equation}
+where $y_c$ is the camber line point at a given value of $x$, $m$ is the
+maximum camber, and $p$ is the location of maximum camber.
+
+\bigbreak
+The upper surface equation is
+\begin{equation}
+    x_u = x - y_h \sin\theta
+    \label{eq:naca:up_x}
+\end{equation}
+\begin{equation}
+    y_u = y_c + y_h \cos\theta
+    \label{eq:naca:up_y}
+\end{equation}
+and the lower surface equation is
+\begin{equation}
+    x_l = x + y_h \sin\theta
+\end{equation}
+\begin{equation}
+    y_l = y_c - y_h \cos\theta
+\end{equation}
+where
+\begin{equation}
+    \theta =
+    \begin{cases}
+    \frac{2m}{p^2} (p - x); & 0 \leq x \leq p \\
+    \frac{2m}{(1-p)^2} (p - x);  &  p \leq x \leq 1
+    \end{cases}
+\end{equation}
+
+\end{flushleft}
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%
+\chapter{Cubic Bezier Curve}
+% ref: https://en.wikipedia.org/wiki/B%C3%A9zier_curve#Cubic_B%C3%A9zier_curves
+
+%%
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+\begin{flushleft}
+
+The cubic Bezier curve is defined by 4 controll points $P_0(X_0, Y_0)$,
```

Comment:
For general multi-dimensional spaces (2D and 3D), $P_{0, 1, 2, 3}$ are vectors.  For simplicity we can use the Cartesian coordinate system to formulate but later we should generalize using a simple vector notation.